source/common/network/utility.cc

```diff
@@ -236,6 +236,9 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
       if (!isLoopbackAddress(*interface_address.interface_addr_) &&
           interface_address.interface_addr_->ip()->version() == version) {
         ret = interface_address.interface_addr_;
+        if (ret->ip()->version() == Address::IpVersion::v6) {
```

Comment:
Is it possible that ret is null, or ret->ip() is null? Maybe:
```
if (ret && ret->ip() && ret->ip()->version() == Address::IpVersion::v6) {
````

Reply:
I don't think the null guard is needed here because in the if statement above, there's already a `interface_address.interface_addr_->ip()->version() == version` call.

Reply:
Oh good lord! You're totally right.